### PR TITLE
Enhance and fix `indentation_linter`

### LINKED ISSUE #1751: indentation_linter thrown off by comment
Is the indentation lint here a false positive? AFAICT, the indentation should be 6 spaces here.

> Indentation should be 4 spaces but is 6 spaces.

Btw, if I try to inline this, I create `line_length_linter()` lint.

__Originally posted by @IndrajeetPatil in https://github.com/r-lib/lintr/pull/1748#discussion_r1002654036__

This should not lint:

```r
a <- # comment
  42L
```

### PR DESCRIPTION
**Pull request by AshesITR**
The issue in the reprex was `xp_block_ends` landed on line 1 because the comment was sitting there. It should have ended on line 2.

### CODE REVIEW

**Comment by AshesITR** [comment]
The R 4.2.2 release scheduled for today seems to interfere with GHA.

**Comment by AshesITR** [comment]
The condition for the double indent is very strict for now (it requires exactly the format specified in the style guide, i.e. no trailing arguments after `(` and an argument before `)`.
This allows us to simultaneously support (single) block-indented function definitions and hanging function definitions:

```r
# all good
function(a = 1,
         b = 2) {
  # body
}

function(
  a = 1,
  b = 2
) {
  # body
}

function(
    a = 1,
    b = 2) {
  # body
}
```

**Comment by AshesITR** [comment]
Filed https://github.com/r-lib/actions/issues/655 for the setup-r problem.

File: tests/testthat/test-indentation_linter.R [hunk 1]
```diff
@@ -163,6 +173,19 @@ test_that("function argument indentation works in tidyverse-style", {
     linter
   )
 
+  # new style (#1754)
```
Comment (MichaelChirico) [inline tests/testthat/test-indentation_linter.R:176]: maybe a few more tests? e.g. triple-indenting gets linted.

does mixing single- and double-indenting lint?

single-indenting should lint under "tidy" too right?

**Review by MichaelChirico — COMMENTED** [review]
I would nolint the cyclocomp issue and file a follow up to keep the logic change here easier to follow

**Comment by AshesITR** [comment]
Took me a few days to get back to it, but factoring out the indentation style XPath logic into small functions made the code easier to digest (for me, at least 🤣) and fixed the cyclocomp lint.

**Review by MichaelChirico — APPROVED** [review]

**Comment by MichaelChirico** [comment]
Great stuff!

**Pull request merged**